### PR TITLE
Initial work for BDC Dashboard

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
This is for just the main BDC overview page. Will be adding the other pages next and continuing to iterate on this - wanted to get what I have in now for early reviewing/commenting.

Note that this is just the backing code for this - I purposely have omitted adding the menu action to launch the dashboard (will be right click from BDC controllers in the Data Explorer view)

The backing data is currently being mocked - will be replaced with real data from the management endpoint once the API issues are resolved. 

UI will continue to be worked on to align with design prototypes. 

image

### CODE REVIEW

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+			});
+
+			contentContainer.addItem(endpointsContainer);
+
+			await view.initializeModel(dashboardRootContainer);
+		});
+	}
+}
+
+function getFriendlyEndpointNames(name: string): string {
```
Comment (kevcunnane) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts]: It's frustrating we have to duplicate this stuff from the mssql extension. Note I'm making changes that'll impact this (there are new endpoints we need to add, and logic on which should be hyperlinked). Can we check with Anthony and see if (in the future, not this PR) we can more easily share some component logic between extensions?

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			{ serviceName: 'Spark', status: { state: 'Ready', healthStatus: 'Healthy' }, resources: undefined },
+			{ serviceName: 'Control', status: { state: 'Ready', healthStatus: 'Healthy' }, resources: undefined },
+			{ serviceName: 'Gateway', status: { state: 'Ready', healthStatus: 'Healthy' }, resources: undefined },
+			{ serviceName: 'App', status: { state: 'Ready', healthStatus: 'Healthy' }, resources: undefined }
+		];
+	}
+
+	public get serviceEndpoints(): { serviceName: string, hyperlink?: string, isHyperlink: boolean, ipAddress?: string, port?: string }[] {
+		return [
+			{ serviceName: 'SQL Server Master Instance', ipAddress: '10.91.134.112', port: '31433', isHyperlink: false },
```
Comment (kevcunnane) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: So: we're moving from ipAddress + port to endpoint, see my new PR. You'll want to update but doesn't need to be done in this review. Just note what I've done in serviceEndpoints.ts in my review for reference.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts]: Agreed. I think the best option for this is to have a core extension (mssql) export functionality like this. That's pretty easy to set up. I'll create an issue for tracking this next release.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: Yup, my local changes are already set up using the new API - this is just pretty old 😄

**Pull request merged**